### PR TITLE
feat: display erc-20 tokens

### PR DESCRIPTION
**Pull request by begonaalvarezd**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

⚠️ To limit the scope of this PR we have not added the real erc-20 token metadata, instead we are using a fake fUSDC token in the meantime until we fetch metadata in the right place, which also needs caching.

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/layer-2/utils/getErc20TokenSymbol.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-console */
+
```
Comment (MarkNerdi) [inline packages/shared/lib/core/layer-2/utils/getErc20TokenSymbol.ts:2]: remove

File: packages/shared/lib/core/layer-2/utils/getErc20TokenSymbol.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { get } from 'svelte/store'
+
+import { network } from '@core/network/stores'
+
+import { ContractType } from '../enums'
+
+export async function getErc20TokenSymbol(tokenAddress: string, chainId: number): Promise<string | undefined> {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/layer-2/utils/getErc20TokenSymbol.ts:9]: do we need an own function for this? this is only used in the ImportErc20Token Popup and there only to display the symbol in the success toast. 
I think we can also use the `getErc20Metadata` function there

File: packages/shared/lib/core/layer-2/utils/getErc20TokenSymbol.ts (hunk 1 above)
Comment (begonaalvarezd) [reply packages/shared/lib/core/layer-2/utils/getErc20TokenSymbol.ts]: pushed

File: packages/shared/lib/core/layer-2/utils/getErc20TokenSymbol.ts (hunk 2 above)
Comment (begonaalvarezd) [reply packages/shared/lib/core/layer-2/utils/getErc20TokenSymbol.ts]: pushed

File: packages/shared/lib/core/wallet/actions/updateActiveAccountTrackedTokens.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+export function updateActiveAccountTrackedTokens(tokenAddress: string, chainId: number): void {
+    let trackedTokens = get(selectedAccount)?.trackedTokens ?? {}
+    const chainIdTrackedTokens = trackedTokens[chainId] ?? []
+    if (!chainIdTrackedTokens.includes(tokenAddress)) {
+        chainIdTrackedTokens.push(tokenAddress)
+        trackedTokens = { ...trackedTokens, [chainId]: chainIdTrackedTokens }
+        updateActiveAccount(get(selectedAccountIndex), { trackedTokens })
+        updateActiveAccountPersistedData(get(selectedAccountIndex), { trackedTokens })
+    }
```
Comment (MarkNerdi) [inline packages/shared/lib/core/wallet/actions/updateActiveAccountTrackedTokens.ts:14]:
```suggestion
    const account = get(selectedAccount)
    if (!account) {
        return
    }
    
    let trackedTokens = account.trackedTokens ?? {}
    const chainIdTrackedTokens = trackedTokens[chainId] ?? []
    if (!chainIdTrackedTokens.includes(tokenAddress)) {
        chainIdTrackedTokens.push(tokenAddress)
        trackedTokens = { ...trackedTokens, [chainId]: chainIdTrackedTokens }
        updateActiveAccount(account.index, { trackedTokens })
        updateActiveAccountPersistedData(account.index, { trackedTokens })
    }
```

File: packages/shared/lib/core/layer-2/actions/fetchSelectedAccountLayer2Balance.ts [hunk 4]
```diff
@@ -72,3 +86,28 @@ async function getSelectedAccountLayer2BalanceForAddress(
...
+        const erc20TokenBalances = []
+        for (const erc20Address of trackedTokens) {
+            const contract = chain?.getContract(ContractType.Erc20, erc20Address)
+            if (contract) {
+                const coinType = chain?.getConfiguration().coinType
+                if (coinType) {
+                    const rawBalance = await contract.methods.balanceOf(selectedAccountEvmAddress).call()
+                    erc20TokenBalances.push({ balance: rawBalance, tokenId: erc20Address })
+                }
+            }
```
Comment (MarkNerdi) [inline packages/shared/lib/core/layer-2/actions/fetchSelectedAccountLayer2Balance.ts:106]: im not a big fan of this nested `if`s. i suggest:

```suggestion
            const contract = chain?.getContract(ContractType.Erc20, erc20Address)
            const coinType = chain?.getConfiguration().coinType
            if (!contract || !coinType) {
                continue
            }
            const rawBalance = await contract.methods.balanceOf(selectedAccountEvmAddress).call()
            erc20TokenBalances.push({ balance: rawBalance, tokenId: erc20Address })
```

File: packages/shared/lib/core/layer-2/actions/fetchSelectedAccountLayer2Balance.ts [hunk 5]
```diff
@@ -45,6 +47,18 @@ export function fetchSelectedAccountLayer2Balance(): void {
 async function getSelectedAccountLayer2BalanceForAddress(
     selectedAccountEvmAddress: string,
     chain: IChain
+): Promise<{ balance: number; tokenId: string }[] | undefined> {
+    const layer2BaseAndIrc30Balances = await getSelectedAccountLayer2NativeTokenBalancesForAddress(
+        selectedAccountEvmAddress,
+        chain
+    )
+    const erc20Balances = await getSelectedAccountLayer2Erc20BalancesForAddress(selectedAccountEvmAddress, chain)
+    return [...(layer2BaseAndIrc30Balances ?? []), ...(erc20Balances ?? [])]
```
Comment (MarkNerdi) [inline packages/shared/lib/core/layer-2/actions/fetchSelectedAccountLayer2Balance.ts:56]: `layer2BaseAndIrc30Balances` is always at least an `[]` so we dont need the `?? [] `

Comment (MarkNerdi) [reply packages/shared/lib/core/layer-2/actions/fetchSelectedAccountLayer2Balance.ts:56]: same for `erc20Balances`

**Review by jeeanribeiro — APPROVED** [review]

**Pull request merged**